MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -93,57 +93,143 @@
+		.variablePointersStorageBuffer = true,
+		.variablePointers = true,
+		.protectedMemory = false,
+		.samplerYcbcrConversion = true,
+		.shaderDrawParameters = true,
+	};
+
+	// Create a SSOT for these Vulkan 1.2 features, which can be queried via two mechanisms here.
+	VkPhysicalDeviceVulkan12Features supportedFeats12 = {
+		.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_VULKAN_1_2_FEATURES,
+		.pNext = nullptr,
+		.samplerMirrorClampToEdge = _metalFeatures.samplerMirrorClampToEdge,
+		.drawIndirectCount = false,									// VK_KHR_draw_indirect_count
+		.storageBuffer8BitAccess = true,
+		.uniformAndStorageBuffer8BitAccess = true,
+		.storagePushConstant8 = true,
+		.shaderBufferInt64Atomics = false,							// VK_KHR_shader_atomic_int64
+		.shaderSharedInt64Atomics = false,							// VK_KHR_shader_atomic_int64
+		.shaderFloat16 = true,
+		.shaderInt8 = true,
+		.descriptorIndexing = false,		// Requires _metalFeatures.arrayOfTextures && _metalFeatures.arrayOfSamplers && shaderStorageBufferArrayNonUniformIndexing
+		.shaderInputAttachmentArrayDynamicIndexing = _metalFeatures.arrayOfTextures,
+		.shaderUniformTexelBufferArrayDynamicIndexing = _metalFeatures.arrayOfTextures,
+		.shaderStorageTexelBufferArrayDynamicIndexing = _metalFeatures.arrayOfTextures,
+		.shaderUniformBufferArrayNonUniformIndexing = false,
+		.shaderSampledImageArrayNonUniformIndexing = _metalFeatures.arrayOfTextures && _metalFeatures.arrayOfSamplers,
+		.shaderStorageBufferArrayNonUniformIndexing = false,
+		.shaderStorageImageArrayNonUniformIndexing = _metalFeatures.arrayOfTextures,
+		.shaderInputAttachmentArrayNonUniformIndexing = _metalFeatures.arrayOfTextures,
+		.shaderUniformTexelBufferArrayNonUniformIndexing = _metalFeatures.arrayOfTextures,
+		.shaderStorageTexelBufferArrayNonUniformIndexing = _metalFeatures.arrayOfTextures,
+		.descriptorBindingUniformBufferUpdateAfterBind = true,
+		.descriptorBindingSampledImageUpdateAfterBind = true,
+		.descriptorBindingStorageImageUpdateAfterBind = true,
+		.descriptorBindingStorageBufferUpdateAfterBind = true,
+		.descriptorBindingUniformTexelBufferUpdateAfterBind = true,
+		.descriptorBindingStorageTexelBufferUpdateAfterBind = true,
+		.descriptorBindingUpdateUnusedWhilePending = true,
+		.descriptorBindingPartiallyBound = true,
+		.descriptorBindingVariableDescriptorCount = true,
+		.runtimeDescriptorArray = true,
+		.samplerFilterMinmax = false,								// VK_EXT_sampler_filter_minmax
+		.scalarBlockLayout = true,
+		.imagelessFramebuffer = true,
+		.uniformBufferStandardLayout = true,
+		.shaderSubgroupExtendedTypes = _metalFeatures.simdPermute || _metalFeatures.quadPermute,
+		.separateDepthStencilLayouts = true,
+		.hostQueryReset = true,
+		.timelineSemaphore = true,
+		.bufferDeviceAddress = mvkOSVersionIsAtLeast(12.05, 16.0),
+		.bufferDeviceAddressCaptureReplay = false,
+		.bufferDeviceAddressMultiDevice = false,
+		.vulkanMemoryModel = false,									// VK_KHR_vulkan_memory_model
```

Comment:
FYI: Sooner or later, we're gonna have to figure out how to support this. Vulkan memory model support is required for Vulkan 1.3.

Reply:
Agreed. Setting these `false` is a precursor for now. We will definitely dig into each of the marked extensions further leading up to a Vulkan 1.2 release (and beyond).